### PR TITLE
fix(formatter): fix else indentation after comment between if-else

### LINKED ISSUE #2649: Problem with newline not being inserted after comments when a token follows it
something like:
```rust
if self == PrimaryColor::Blue {
            Rgb {
                red: 0,
                blue: 255,
                green: 0,
            }
        }
        // TODO remove this else when exhaustive ifs are checked for
      else {
            Rgb {
                red: 0,
                green: 0,
                blue: 0,
            }
        }
```
turns into:
```rust
if self == PrimaryColor::Blue {
            Rgb {
                red: 0,
                blue: 255,
                green: 0,
            }
        }
        // TODO remove this else when exhaustive ifs are checked for else { <-- part of the comment now, should be on newline
            Rgb {
                red: 0,
                green: 0,
                blue: 0,
            }
        }
```

- [x] #2654

### PR DESCRIPTION
**Pull request by bingcicle**
closes #2649

Added a few test cases with comments in between, including:
- single-line comment with overindentation,
- multi-line comment with underindentation,
- single-line comment, from inline to multiline if-else

Updated `get_post_context()` with a better way to format `else` after comment. This is a bit of a hacky fix but since the post-context concept is currently only used for formatting `else` blocks after comment, I was thinking this might tide us through for now (and the test suite passes as well).

I wrote a long-form comment explaining what the new get_post_context() does there especially because fixing the off-by-one error with subtracting by 1 is going to look like a hacky solution (and it is, unfortunately)

### CODE REVIEW

**Review by kayagokalp — APPROVED** [review]
utACK ✅

**Comment by bingcicle** [comment]
So this doesn't actually fix the case where the comments themselves are unformatted. Example:

```rust
contract;

impl MyContract for Contract {
    fn is_blue() -> bool {
        if self == PrimaryColor::Blue {
            true
        }
        // TODO
            // remove this else when exhaustive ifs are checked for
    else {
            false
        }
    }
}"#;
```

I'd say this PR is still worth merging since it fixes the linked issue, and I guess misindented comment would be rarer than a misindented `else` block. I'll open a new issue for the above.

File: swayfmt/src/formatter/mod.rs [hunk 1]
```diff
@@ -1375,6 +1375,115 @@ fn foo() {}
...
+    #[test]
+    fn test_comment_between_if_else_overindented() {
+        let sway_code_to_format = r#"contract;
+
+impl MyContract for Contract {
+    fn is_blue() -> bool {
+        if self == PrimaryColor::Blue {
+            true
+        }
+        // TODO remove this else when exhaustive ifs are checked for
```
Comment (sdankel) [inline swayfmt/src/formatter/mod.rs:1395]: It seems like the new logic is assuming that the comment is correctly aligned. Do we have a test case where both the comment and the else block are misaligned? Something like:

```
impl MyContract for Contract {
    fn is_blue() -> bool {
        if self == PrimaryColor::Blue {
            true
        }
                      // TODO remove this else when exhaustive ifs are checked for
                else {
            false
        }
    }
}"#;
```

**Review by sdankel — APPROVED** [review]

File: swayfmt/src/formatter/mod.rs (hunk 1 above)
Comment (bingcicle) [reply swayfmt/src/formatter/mod.rs:1395]: Can confirm that this is still an issue, output:

```rust
impl MyContract for Contract {
    fn is_blue() -> bool {
        if self == PrimaryColor::Blue {
            true
        }
                // TODO remove this else when exhaustive ifs are checked for
                else {
            false
        }
    }
}"#;
```

**Pull request merged**